expo/expo
# [router] wraps return of useLocalSearchParams with useMemo

# Why

Currently, when using const params = useLocalSearchParams() within other hooks, these hooks are being called on every component render. This can lead to performance issues, especially in scenarios where the params are used in effects with API calls.

```
const params = useLocalSearchParams();

useEffect(() => {
  // This effect runs on every render due to params changing
  fetchData(params);
}, [params]);
```

# How

Wrapping return of `useLocalSearchParams` with `React.useMemo`

# Test Plan

To test this change:

1. Create a component that uses `const params = useLocalSearchParams` and a dependent effect or hook.
2. Add console logs in the effect/hook to track executions.
3. Navigate to the component with query parameters and interact with it to trigger rerenders.
4. Before fix: Observe frequent console logs, indicating unnecessary effect/hook executions.
5. After fix: Verify that console logs only appear when search params actually change.

# Checklist
